lib/src/extended_resize_image_provider.dart

```diff
@@ -0,0 +1,231 @@
+import 'dart:async';
+import 'dart:math';
+import 'dart:typed_data';
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import 'extended_image_provider.dart';
+
+/// Instructs Flutter to decode the image at the specified dimensions
+/// instead of at its native size.
+///
+/// This allows finer control of the size of the image in [ImageCache] and is
+/// generally used to reduce the memory footprint of [ImageCache].
+///
+/// The decoded image may still be displayed at sizes other than the
+/// cached size provided here.
+///
+/// [scaling] If not null, the original image's width and height will
+/// divide with it to calculate targetWidth.
+/// [maxBytes] If not null, the image will resize to a size that nolarger
+/// than [maxBytes]. Default size is 500KB.
+class ExtendedResizeImage extends ImageProvider<_SizeAwareCacheKey>
+    with ExtendedImageProvider<_SizeAwareCacheKey> {
+  const ExtendedResizeImage(
+    this.imageProvider, {
+    this.scaling,
+    this.maxBytes = 500 << 10,
+    this.width,
+    this.height,
+    this.allowUpscaling = false,
+  })  : assert(scaling != null ||
```

Comment:
Split the assertion to single field assertions, except the `(width != null || height != null)`.